### PR TITLE
Add unit testing framework

### PR DESCRIPTION
**Pull request by DerekTang04**
First pass. May be very unhinged.

### CODE REVIEW

File: .github/workflows/test_action.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: GitHub Actions Demo
```
Comment (SnackkOverflowError) [inline .github/workflows/test_action.yaml:1]: Do we need this?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+## Description
```
Comment (SnackkOverflowError) [inline .github/pull_request_template.md:1]: Ideally this is its own PR but its okay this time.

File: AttitudeManager/Inc/temp.txt
Comment (SnackkOverflowError) [inline AttitudeManager/Inc/temp.txt:1]: I think you can remove all these temp.txt Theyre just gonna get deleted when someone pushes to this folder anyway. We can enforce the naming of 'Inc' and 'Src' without these

File: Testing/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+set(CMAKE_C_COMPILER "gcc")
+set(CMAKE_CXX_COMPILER "g++")
+
+# GoogleTest requires at least C++14
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Locate GTest
+find_package(GTest REQUIRED)
```
Comment (SnackkOverflowError) [inline Testing/CMakeLists.txt:12]: Could you real quick make a 'unit testing for zp3.5' doc under ZeroPilot 3.5 on confluence. You dont need to add anything to it yet, but maybe just a link to where gtest can be installed?

File: Testing/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+)
+
+# --- ZeroPilot File Indexing End ---
+
+add_executable(test_build
+    ${AM_CODE}
+    ${PM_CODE}
+    ${SM_CODE}
+    ${TM_CODE}
+)
```
Comment (SnackkOverflowError) [inline Testing/CMakeLists.txt:81]: add drivers cpp files here, the same way the managers were done

File: Drivers/Tests/temp.txt
Comment (SnackkOverflowError) [inline Drivers/Tests/temp.txt:1]: can you make a test "hello world" unit test and put it in this folder?

**Review by SnackkOverflowError — CHANGES_REQUESTED** [review]
overall looks good. small changes

File: .github/workflows/test_action.yaml (hunk 1 above)
Comment (DerekTang04) [reply .github/workflows/test_action.yaml]: No, that was for my testing, and I forgot about it. Removed now.

File: .github/pull_request_template.md (hunk 2 above)
Comment (DerekTang04) [reply .github/pull_request_template.md:1]: Noted.

File: Drivers/Tests/temp.txt
Comment (DerekTang04) [reply Drivers/Tests/temp.txt:1]: done

File: Testing/CMakeLists.txt (hunk 4 above)
Comment (DerekTang04) [reply Testing/CMakeLists.txt:95]: added. also fixed an error in the unit test CMakeLists

File: AttitudeManager/Inc/temp.txt
Comment (DerekTang04) [reply AttitudeManager/Inc/temp.txt:1]: removed all the .txt files. only ones left are from chris's inital file setup.

File: Testing/CMakeLists.txt (hunk 3 above)
Comment (DerekTang04) [reply Testing/CMakeLists.txt:12]: done

**Review by SnackkOverflowError — APPROVED** [review]
LGTM

**Review by Chrisc110 — APPROVED** [review]

**Pull request merged**